Source/objects.cpp

```diff
@@ -4564,33 +4564,33 @@ bool OperateFountains(int pnum, int i)
 			return false;
 		if (pnum != myplr)
 			return false;
-		int prev = -1;
-		int add = -1;
-		int cnt = 0;
-		while (true) {
-			int rnd = GenerateRnd(4);
-			if (rnd != prev) {
-				switch (rnd) {
-				case 0:
-					ModifyPlrStr(pnum, add);
-					break;
-				case 1:
-					ModifyPlrMag(pnum, add);
-					break;
-				case 2:
-					ModifyPlrDex(pnum, add);
-					break;
-				case 3:
-					ModifyPlrVit(pnum, add);
-					break;
-				}
-				prev = rnd;
-				add = 1;
-				cnt++;
-			}
-			if (cnt > 1)
+
+		int fromStat = GenerateRnd(4);
+		int toStat = [&]() {
+			int rnd = GenerateRnd(3);
+			if (rnd >= fromStat)
+				rnd++;
+			return rnd;
+		}();

```

Comment:
What is the lambda giving us here compared to just dowing it directly?
```suggestion
		int toStat = GenerateRnd(3);
		if (toStat >= fromStat)
			toStat++;
```

Reply:
Slightly better indication of intent, but otherwise, nothing.

Reply:
Imo it adds a significant number of cryptic symbols that some new comers may not be familiar with, for something like level objects I think it's important that the code be very approachable for novices.

Reply:
Alright.

I prefer the lambda because toStat >= fromStat isn't an exceptional condition.